backend/channels.py

```diff
@@ -621,14 +736,40 @@ async def publish_channel_muxes(config):
                     else:
                         logger.info("    - Updating existing MUX '%s' for channel '%s' with stream '%s'", mux_uuid,
                                     result.name, source.playlist_stream_url)
+                    
+                    # Determine EPG ID to use
+                    epg_id = None
+                    if result.guide_channel_id:
+                        epg_id = result.guide_channel_id
+                        logger.info(f"    - Using configured EPG ID: {epg_id}")
+                    else:
+                        # Versuche EPG-ID aus Kanalname oder anderen Quellen zu finden
+                        normalized_name = re.sub(r'[^a-zA-Z0-9]', '', result.name.lower())
+                        for epg_ch_id, epg_data in epg_channels.items():
+                            normalized_epg_name = re.sub(r'[^a-zA-Z0-9]', '', epg_data['name'].lower())
+                            if normalized_name == normalized_epg_name or normalized_name in normalized_epg_name:
+                                epg_id = epg_ch_id
+                                # Aktualisiere auch den Kanal mit dieser EPG-ID
+                                result.guide_channel_id = epg_id
+                                result.guide_id = epg_data['epg_id']
+                                result.guide_name = epg_data['name']
+                                db.session.commit()
```

Comment:
I think adding a section to update the channel data stored in the database here in this function that is publishing the MUX to the TVH backend is wrong.
The mapping of a channel to the EPG should only be done in the "Programme Guide:" section when editing a channel in the UI. This is a user-controlled feature that they map an EPG channel to a collection of M3U streams. Adding this code here does not make sense to me. I feel like if a user does not configure a programme guide for their channel, then it should be left blank.